e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/filter-payment-details.spec.js

```diff
@@ -26,52 +28,693 @@ context(`users can filter payment details by facility id and payment reference a
     cy.task(NODE_TASKS.REMOVE_ALL_UTILISATION_REPORTS_FROM_DB);
 
     cy.task(NODE_TASKS.INSERT_UTILISATION_REPORTS_INTO_DB, [utilisationReport]);
+  });
+
+  describe('when filter panel toggle button is clicked', () => {
+    it('should toggle the filter panel', () => {
+      const feeRecord = FeeRecordEntityMockBuilder.forReport(utilisationReport).withId(1).build();
+      const payment = PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP).withFeeRecords([feeRecord]).build();
+
+      cy.task(NODE_TASKS.INSERT_FEE_RECORDS_INTO_DB, [feeRecord]);
+      cy.task(NODE_TASKS.INSERT_PAYMENTS_INTO_DB, [payment]);
+
+      pages.landingPage.visit();
+      cy.login(USERS.PDC_RECONCILE);
+
+      cy.visit(`/utilisation-reports/${reportId}`);
+
+      pages.utilisationReportPage.paymentDetailsTabLink().click();
+
+      // On first page load, the filter panel should be visible.
+      pages.utilisationReportPage.paymentDetailsTab.filters.panel().should('be.visible');
+
+      cy.assertText(pages.utilisationReportPage.paymentDetailsTab.filters.panelToggleButton(), 'Hide filters');
+      pages.utilisationReportPage.paymentDetailsTab.filters.panelToggleButton().click();
+      pages.utilisationReportPage.paymentDetailsTab.filters.panel().should('not.be.visible');
+
+      cy.assertText(pages.utilisationReportPage.paymentDetailsTab.filters.panelToggleButton(), 'Show filters');
+      pages.utilisationReportPage.paymentDetailsTab.filters.panelToggleButton().click();
+      pages.utilisationReportPage.paymentDetailsTab.filters.panel().should('be.visible');
```

Comment:
Could we destructure so easier to read? E.g

```js
filters.panel().should('be.visible');

cy.assertText(filters.panelToggleButton(), 'Hide filters');

filters.panelToggleButton().click();
filters.panel().should('not.be.visible');

cy.assertText(filters.panelToggleButton(), 'Show filters');

filters.panelToggleButton().click();
filters.panel().should('be.visible');
```

Reply:
Actually if we could destructure all of the selectors in this test (to some degree so it still makes sense) I think the test will be easier to read/maintain, lots of long lines atm e.g

```js
pages.utilisationReportPage.paymentDetailsTab.paymentDetailsTable.row(firstPayment.id, firstFeeRecord.id).should('exist');
pages.utilisationReportPage.paymentDetailsTab.paymentDetailsTable.row(secondPayment.id, secondFeeRecord.id).should('not.exist');
```